packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit_api_impls.dart

```diff
@@ -0,0 +1,511 @@
+        value = WKWebsiteDataTypesEnum.sessionStorage;
+        break;
+      case WKWebsiteDataTypes.sqlDatabases:
+        value = WKWebsiteDataTypesEnum.sqlDatabases;
+        break;
+      case WKWebsiteDataTypes.indexedDBDatabases:
+        value = WKWebsiteDataTypesEnum.indexedDBDatabases;
+        break;
+    }
+
+    return WKWebsiteDataTypesEnumData(value: value);
+  });
+}
+
+extension _WKUserScriptInjectionTimeConverter on WKUserScriptInjectionTime {
+  WKUserScriptInjectionTimeEnumData toWKUserScriptInjectionTimeEnumData() {
+    late final WKUserScriptInjectionTimeEnum value;
+    switch (this) {
+      case WKUserScriptInjectionTime.atDocumentStart:
+        value = WKUserScriptInjectionTimeEnum.atDocumentStart;
+        break;
+      case WKUserScriptInjectionTime.atDocumentEnd:
+        value = WKUserScriptInjectionTimeEnum.atDocumentEnd;
+        break;
+    }
+
+    return WKUserScriptInjectionTimeEnumData(value: value);
+  }
+}
+
+Iterable<WKAudiovisualMediaTypeEnumData> _toWKAudiovisualMediaTypeEnumData(
+  Iterable<WKAudiovisualMediaType> types,
+) {
+  return types
+      .map<WKAudiovisualMediaTypeEnumData>((WKAudiovisualMediaType type) {
+    late final WKAudiovisualMediaTypeEnum value;
+    switch (type) {
+      case WKAudiovisualMediaType.none:
+        value = WKAudiovisualMediaTypeEnum.none;
+        break;
+      case WKAudiovisualMediaType.audio:
+        value = WKAudiovisualMediaTypeEnum.audio;
+        break;
+      case WKAudiovisualMediaType.video:
+        value = WKAudiovisualMediaTypeEnum.video;
+        break;
+      case WKAudiovisualMediaType.all:
+        value = WKAudiovisualMediaTypeEnum.all;
+        break;
+    }
+
+    return WKAudiovisualMediaTypeEnumData(value: value);
+  });
+}
+
+extension _WKUserScriptConverter on WKUserScript {
+  WKUserScriptData toWKUserScriptData() {
+    return WKUserScriptData(
+      source: source,
+      injectionTime: injectionTime.toWKUserScriptInjectionTimeEnumData(),
+      isMainFrameOnly: isMainFrameOnly,
+    );
+  }
+}
+
+extension _NSUrlRequestConverter on NSUrlRequest {
+  NSUrlRequestData toNSUrlRequestData() {
+    return NSUrlRequestData(
+      url: url,
+      httpMethod: httpMethod,
+      httpBody: httpBody,
+      allHttpHeaderFields: allHttpHeaderFields,
+    );
+  }
+}
+
+/// Host api implementation for [WKWebSiteDataStore].
+class WKWebsiteDataStoreHostApiImpl extends WKWebsiteDataStoreHostApi {
+  /// Constructs a [WebsiteDataStoreHostApiImpl].
+  WKWebsiteDataStoreHostApiImpl({
+    BinaryMessenger? binaryMessenger,
+    InstanceManager? instanceManager,
+  })  : instanceManager = instanceManager ?? InstanceManager.instance,
+        super(binaryMessenger: binaryMessenger);
+
+  /// Maintains instances stored to communicate with Objective-C objects.
+  final InstanceManager instanceManager;
+
+  /// Converts objects to instances ids for [createFromWebViewConfiguration].
+  Future<void> createFromWebViewConfigurationFromInstance(
```

Comment:
The From/From naming is kind of awkward; would `WithInstance` be better for all of these maybe? Or `ForInstance`?

Reply:
I went with `ForInstances`